### ISSUE TITLE
Re-Import of Calendar fails

### ISSUE THREAD

**Issue by kmorgen1** [nextcloud/calendar#780]
### Steps to reproduce
1. Export a calendar from Nextcloud (one that was imported with errors from an owncloud-generated ics-file)
2. Import the same calendar into Nextcloud

### Expected behaviour
Import with no errors

### Actual behaviour
Import with some entries failing to import (number of entries failling to import varies between tries)

## Server configuration detail
**Operating system:** Linux 3.13.0-042stab127.2 #1 SMP Thu Jan 4 16:41:44 MSK 2018 x86_64

**Webserver:** Apache (cgi-fcgi)

**Database:** mysql 5.5.59

**PHP version:** 7.0.28
Modules loaded: Core, date, libxml, openssl, pcre, zlib, bz2, calendar, ctype, hash, filter, ftp, gettext, gmp, SPL, iconv, pcntl, readline, Reflection, session, standard, SimpleXML, sockets, mbstring, tokenizer, xml, cgi-fcgi, mysqlnd, bcmath, curl, dba, dom, enchant, fileinfo, gd, imagick, imap, intl, json, ldap, exif, mcrypt, mysqli, odbc, PDO, pdo_mysql, PDO_ODBC, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, pspell, redis, soap, sqlite3, sysvmsg, sysvsem, sysvshm, tidy, xmlreader, xmlrpc, xmlwriter, xsl, zip, ionCube Loader, Zend OPcache

**Nextcloud version:** 13.0.1 - 13.0.1.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from a fresh install auf 13.0

**Where did you install Nextcloud from:** downloaded zip from Nextcloud.com

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.6.1
 - calendar: 1.6.1
 - comments: 1.3.0
 - contacts: 2.1.2
 - dav: 1.4.6
 - federatedfilesharing: 1.3.1
 - federation: 1.3.0
 - files: 1.8.0
 - files_pdfviewer: 1.2.0
 - files_sharing: 1.5.0
 - files_texteditor: 2.5.1
 - files_trashbin: 1.3.0
 - files_versions: 1.6.0
 - files_videoplayer: 1.2.0
 - firstrunwizard: 2.2.1
 - gallery: 18.0.0
 - issuetemplate: 0.3.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.1.0
 - nextcloud_announcements: 1.2.0
 - notifications: 2.1.2
 - oauth2: 1.1.0
 - password_policy: 1.3.0
 - provisioning_api: 1.3.0
 - serverinfo: 1.3.0
 - sharebymail: 1.3.0
 - survey_client: 1.1.0
 - systemtags: 1.3.0
 - theming: 1.4.1
 - twofactor_backupcodes: 1.2.3
 - updatenotification: 1.3.0
 - workflowengine: 1.3.0
Disabled:
 - admin_audit
 - encryption
 - files_external
 - user_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "box.kaimorgenstern.net"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/box.kaimorgenstern.net",
    "dbtype": "mysql",
    "version": "13.0.1.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "nc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "htaccess.RewriteBase": "\/",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "memcache.locking": "OC\\Memcache\\Redis",
    "memcache.local": "OC\\Memcache\\Redis",
    "maintenance": false,
    "updater.secret": "***REMOVED SENSITIVE VALUE***",
    "theme": "",
    "loglevel": 2,
    "mail_smtpmode": "smtp",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpsecure": "ssl",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauth": 1,
    "mail_smtpport": "465",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

## Client configuration

**Browser:** Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:59.0) Gecko/20100101 Firefox/59.0

**Operating system:** Ubuntu 14.04

## Logs

<details><summary>Webserver log</summary>

```
Sat Mar 17 23:57:59 2018 (22258): Fatal Error Unable to allocate shared memory segment of 134217728 bytes: mmap: Cannot allocate memory (12)
Sat Mar 17 23:58:01 2018 (22260): Fatal Error Unable to allocate shared memory segment of 134217728 bytes: mmap: Cannot allocate memory (12)
Sat Mar 17 23:59:36 2018 (22279): Fatal Error Unable to allocate shared memory segment of 134217728 bytes: mmap: Cannot allocate memory (12)
Sat Mar 17 23:59:37 2018 (22280): Fatal Error Unable to allocate shared memory segment of 134217728 bytes: mmap: Cannot allocate memory (12)
```
</details>

<details><summary>Nextcloud log</summary>

```
No entries in the Nextcloud log
```
</details>

<details><summary>Browser log</summary>

This appear when navigating to the calendar. No entries during the actual import. However, I am not familiar with using the firefox web-console. Am I missing something? 

Navigated to https://box.kaimorgenstern.net/apps/calendar/
JQMIGRATE: Migrate is installed, version 1.4.0
core.js:7:542
Deprecation warning: use moment.updateLocale(localeName, config) to change an existing locale. moment.defineLocale(localeName, config) should only be used for creating a new locale See http://momentjs.com/guides/#/warnings/define-locale/ for more info.
core.js:1563:2834
Attempting to call a FullCalendar method on an element with no calendar.
vendor.min.js:10:21215
Source-Map-Fehler: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Ressourcen-Adresse: https://box.kaimorgenstern.net/apps/calendar/js/public/vendor.min.js?v=4eafe9e9-0
Source-Map-Adresse: vendor.min.js.map[Weitere Informationen]

</details>

**Comment by georgehrke** [comment]
Can you please check the network log of your browser?

There must be failed connections. Can you check what kind of errors there are? 

Actual error messages or failed connections because there are too many connections?

**Comment by kmorgen1** [comment]
Importing a calendar with one calendar item failing to import the network log of firefox shows one error

500 Internal Server Error
with a time stamp of Sun, 18 Mar 2018 14:57:03 GMT

The corresponding server error log entries are

vhost error log:
```
[Sun Mar 18 15:57:04.553430 2018] [fcgid:warn] [pid 24267] (104)Connection reset by peer: [client 93.234.255.82:53216] mod_fcgid: error reading data from FastCGI server
[Sun Mar 18 15:57:04.553495 2018] [core:error] [pid 24267] [client 93.234.255.82:53216] End of script output before headers: remote.php
```
 /var/log/apache2/error.log:
```
Sun Mar 18 15:57:04 2018 (24268): Fatal Error Unable to allocate shared memory segment of 134217728 bytes: mmap: Cannot allocate memory (12)
```

**Comment by kmorgen1** [comment]
I think I fixed it: Since the error logs mentioned FastCGI I rembered that in the Plesk management panel that I use to configre the domain settings I chose to run PHP 7.0.28 as a FastCGI application. It gives the option of switching it to an FPM application, so I did that and after that I got no more errors on any calendar imports.

Now I wonder whether it was stupid in the first place to combine PHP 7.0.28 with FastCGI or if this could have happened to anyone. If you have any hints on that I'd appreciate it!
Thanks for taking the time to look at something that was clearly my configuration error!

Cheers
kai*

**Comment by georgehrke** [comment]
> Now I wonder whether it was stupid in the first place to combine PHP 7.0.28 with FastCGI or if this could have happened to anyone (PHP 5.5.38 & FastCGI worked for my old owncloud 9.0 install) . If you have any hints on that I'd appreciate it!

It should work with either.

> Sun Mar 18 15:57:04 2018 (24268): Fatal Error Unable to allocate shared memory segment of 134217728 bytes: mmap: Cannot allocate memory (12

This looks like your server ran out of memory. This could very likely (but doesn't have to) be related to the fact that at the moment we sent all the requests for import at the same time and depending on how the server deals with incoming connections it can cause this issues.
We are planning to limit the number of concurrent requests during import: https://github.com/nextcloud/calendar/issues/445

As the actual issue is fixed for you I will close this as a duplicate of #445 for now. If it's not fixed with Calendar 1.7.0 in the future we can always reopen this ticket :)
